### PR TITLE
CMR-9797: es-lib-migration

### PR DESCRIPTION
**Pull request by jceaser**
# Overview

This is the second draft pull request. The first one confused GitHub. Sorry.

### What is the feature/fix?

This change is a migration of search code related to elastic query moved into elastic-util under a "search" directory there. Indexer was not moved due to the already expanding size of this request and will be handled in a future request.

### What is the Solution?
elastic-utils contains a new package called "search" which contains code related to "searching" elastic. It is not just for the search-app as access-control-app has had some files moved in as well.

You will also find a metric ton of linter changes 

### What areas of the application does this impact?

Both search-app and access-control-app have been impacted

# Checklist

- [x] I have updated/added unit and integration tests that prove my fix is effective or that my feature works
- [x] New and existing unit and int tests pass locally and remotely with my changes
- [x] I have removed unnecessary/dead code and imports in files I have changed
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: common-lib/project.clj
Comment (jmaeng72) [inline common-lib/project.clj:68]: Just putting a comment here to remind you to do this before merging

File: common-lib/src/cmr/common/rebalancing_collections.clj
Comment (jmaeng72) [inline common-lib/src/cmr/common/rebalancing_collections.clj:7]: Can you add the ticket number of when this work will be done? I found this helpful for other comments I've seen as a dev coming from outside this work

File: elastic-utils-lib/project.clj
Comment (jmaeng72) [inline elastic-utils-lib/project.clj:38]: reminder to remove before merging

File: elastic-utils-lib/src/cmr/elastic_utils/search/es_index_name_cache.clj
Comment (jmaeng72) [inline elastic-utils-lib/src/cmr/elastic_utils/search/es_index_name_cache.clj:68]: Are you going to return to this?

File: search-app/src/cmr/search/results_handlers/atom_spatial_results_handler.clj
Comment (jmaeng72) [inline search-app/src/cmr/search/results_handlers/atom_spatial_results_handler.clj:5]: why not remove, instead of commenting out?

File: elastic-utils-lib/src/cmr/elastic_utils/search/es_index_name_cache.clj
Comment (eereiter) [reply elastic-utils-lib/src/cmr/elastic_utils/search/es_index_name_cache.clj:68]: remove the TODO.  The defjob is not a variable I think that is why you can't declare it.  I am not sure what it is though - I am anticipating your next question. :-)

File: search-app/src/cmr/search/results_handlers/atom_spatial_results_handler.clj
Comment (eereiter) [reply search-app/src/cmr/search/results_handlers/atom_spatial_results_handler.clj:5]: +1

**Review by eereiter — APPROVED** [review]
I approved it assuming that you will work Jyna's comments.

File: search-app/src/cmr/search/services/query_walkers/temporal_range_extractor.clj
Comment (jmaeng72) [inline search-app/src/cmr/search/services/query_walkers/temporal_range_extractor.clj:6]: remove comment if not using

File: elastic-utils-lib/src/cmr/elastic_utils/config.clj
Comment (jmaeng72) [inline elastic-utils-lib/src/cmr/elastic_utils/config.clj:66]: it does look exactly the same as the defconfig in index-set and elastic_search_index. Unsure if it's a value that can be universal, or if the values change per app though

File: common-lib/project.clj
Comment (jceaser) [reply common-lib/project.clj:68]: thanks, I actually have a couple of others too.

File: common-lib/src/cmr/common/rebalancing_collections.clj
Comment (jceaser) [reply common-lib/src/cmr/common/rebalancing_collections.clj:7]: I actually have one now, so yes

File: elastic-utils-lib/src/cmr/elastic_utils/search/es_index_name_cache.clj
Comment (jceaser) [reply elastic-utils-lib/src/cmr/elastic_utils/search/es_index_name_cache.clj:68]: I think the new config file will take care of this, otherwise, it will just be fixed next time we do a lint

File: search-app/src/cmr/search/services/query_walkers/temporal_range_extractor.clj
Comment (jceaser) [reply search-app/src/cmr/search/services/query_walkers/temporal_range_extractor.clj:6]: this was a test to see if the run time would complain about it, if it did, I would have moved it to the import section.

File: elastic-utils-lib/src/cmr/elastic_utils/config.clj
Comment (jceaser) [reply elastic-utils-lib/src/cmr/elastic_utils/config.clj:66]: the code already had redundant values, they are the same and they will be used by all three blocks of code together, I think these exist this way so that you don't need to share libraries just to get a defconf, I'm just pointing out the oddity in case you want to rename these, I'll add more notes there.

File: search-app/src/cmr/search/results_handlers/atom_spatial_results_handler.clj
Comment (jceaser) [reply search-app/src/cmr/search/results_handlers/atom_spatial_results_handler.clj:5]: I was not sure if it needed to be a require or an import.

**Review by jmaeng72 — APPROVED** [review]

**Pull request merged**
